server/api.js

```diff
@@ -587,6 +588,17 @@ router.post("/bid", async (req, res) => {
 		try {
 			await client.query("BEGIN");
 
+			const checkBidQuery = `
+				SELECT * FROM bid WHERE tender_id = $1 AND bidder_id = $2
+			`;
+			const checkBidValues = [tenderId, bidderId];
+			const existingBid = await client.query(checkBidQuery, checkBidValues);
+
+			if (existingBid.rows.length > 0) {
+				await client.query("ROLLBACK");
+				return res.status(400).json({});
```

Comment:
```suggestion
				return res.status(400).json({code: "DUPLICATE_ENTRY"});
```

Or something similar, so the client can handle the error